packages/opentelemetry-resource-detector-aws/src/detectors/AwsEksDetector.ts

```diff
@@ -0,0 +1,227 @@
+      },
+      ca: cert,
+    };
+    return !!(await this._fetchString(options));
+  }
+
+  /**
+   * Attempts to make a connection to Amazon Cloudwatch
+   * Config Maps to grab cluster name
+   * @param config The resource detection config with a required logger
+   */
+  private async _getClusterName(
+    config: ResourceDetectionConfigWithLogger,
+    cert: Buffer
+  ): Promise<string | undefined> {
+    const options = {
+      host: this.K8S_SVC_URL,
+      path: this.CW_CONFIGMAP_PATH,
+      method: 'GET',
+      timeout: this.MILLISECOND_TIME_OUT,
+      headers: {
+        Authorization: await this._getK8sCredHeader(config),
+      },
+      ca: cert,
+    };
+    const response = await this._fetchString(options);
+    try {
+      return JSON.parse(response).data['cluster.name'];
+    } catch (e) {
+      config.logger.warn('Cannot get cluster name on EKS', e);
+    }
+    return '';
+  }
+  /**
+   * Reads the Kubernetes token path and returns kubernetes
+   * credential header
+   * @param config The resource detection config with a required logger
+   */
+  private async _getK8sCredHeader(
+    config: ResourceDetectionConfigWithLogger
+  ): Promise<string> {
+    try {
+      const content = await AwsEksDetector.readFileAsync(
+        this.K8S_TOKEN_PATH,
+        'utf8'
+      );
+      return 'Bearer ' + content;
+    } catch (e) {
+      config.logger.warn('Unable to read Kubernetes client token.', e);
+    }
+    return '';
+  }
+
+  /**
+   * Read container ID from cgroup file
+   * In EKS, even if we fail to find target file
+   * or target file does not contain container ID
+   * we do not throw an error but throw warning message
+   * and then return null string
+   */
+  private async _getContainerId(
+    config: ResourceDetectionConfigWithLogger
+  ): Promise<string | undefined> {
+    try {
+      const rawData = await AwsEksDetector.readFileAsync(
+        this.DEFAULT_CGROUP_PATH,
+        'utf8'
+      );
+      const splitData = rawData.trim().split('\n');
+      for (const str of splitData) {
+        if (str.length > this.CONTAINER_ID_LENGTH) {
+          return str.substring(str.length - this.CONTAINER_ID_LENGTH);
+        }
+      }
+    } catch (e) {
+      config.logger.warn(
+        `AwsEksDetector failed to read container ID: ${e.message}`
+      );
+    }
+    return undefined;
+  }
+
+  /**
+   * Establishes an HTTP connection to AWS instance document url.
+   * If the application is running on an Eks instance, we should be able
+   * to get back a valid JSON document. Parses that document and stores
+   * the identity properties in a local map.
+   */
+  private async _fetchString(options: https.RequestOptions): Promise<string> {
+    return await new Promise((resolve, reject) => {
+      const timeoutId = setTimeout(() => {
+        req.abort();
+        reject(new Error('Eks metadata api request timed out.'));
+      }, 2000);
+
+      const req = https.request(options, res => {
```

Comment:
This might get instrumented and I think it should not, please verify and see if http plugin will create a span for that which will be then exported. If that is the case then you should use `suppressInstrumentation` from `api`

Reply:
Tested - http plugin does not create an unexpected span which will be exported.